### PR TITLE
Finished writing database schema DDL

### PR DESCRIPTION
**Pull request by CalebMathers**
For issue #72 

Wrote all the tables from the ERD into DDL SQL in a file `schema.sql`, within the `database` directory.

Working on a local Postgres database.

### CODE REVIEW

File: database/schema.sql [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+CREATE TABLE aurora_alert (
+    alert_id BIGINT GENERATED ALWAYS AS IDENTITY,
+    alert_time TIMESTAMP NOT NULL,
+    aurora_colour_id SMALLINT NOT NULL,
+    PRIMARY KEY (alert_id),
+    FOREIGN KEY (aurora_colour_id) REFERENCES aurora_colour(aurora_colour_id)
+);
+
+CREATE TABLE region (
+    region_id SMALLINT,
```
Comment (AlexOBarnes) [inline database/schema.sql:28]:
```suggestion
    region_id SMALLINT GENERATED ALWAYS AS IDENTITY,
```

File: database/schema.sql [hunk 2]
```diff
@@ -0,0 +1,112 @@
+DROP TABLE IF EXISTS subscriber_county_assignment, subscriber, forecast, solar_feature, county, body_assignment, body, constellation, region, aurora_alert, aurora_colour, image;
```
Comment (MurasatoNaoya) [inline database/schema.sql:1]: Dropped tables when the sql script is run, that the reset.zsh script will run later. LGTM.

File: database/schema.sql [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+CREATE TABLE region (
+    region_id SMALLINT,
+    region_name VARCHAR(25) NOT NULL,
+    latitude FLOAT NOT NULL,
+    longitude FLOAT NOT NULL,
+    PRIMARY KEY (region_id)
+);
+
+CREATE TABLE constellation (
+    constellation_id SMALLINT,
```
Comment (AlexOBarnes) [inline database/schema.sql:36]:
```suggestion
    constellation_id SMALLINT GENERATED ALWAYS AS IDENTITY,
```

File: database/schema.sql [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    constellation_name VARCHAR(35) NOT NULL,
+    PRIMARY KEY (constellation_id)
+);
+
+CREATE TABLE body (
+    body_id BIGINT,
```
Comment (AlexOBarnes) [inline database/schema.sql:42]:
```suggestion
    body_id INT GENERATED ALWAYS AS IDENTITY,
```

File: database/schema.sql [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    end_time TIMESTAMP NOT NULL,
+    latitude FLOAT NOT NULL,
+    longitude FLOAT NOT NULL,
+    PRIMARY KEY (assignment_id),
+    FOREIGN KEY (region_id) REFERENCES region(region_id),
+    FOREIGN KEY (body_id) REFERENCES body(body_id)
+);
+
+CREATE TABLE county (
+    county_id SMALLINT,
```
Comment (AlexOBarnes) [inline database/schema.sql:63]:
```suggestion
    county_id SMALLINT GENERATED ALWAYS AS IDENTITY,
```

File: database/schema.sql [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    cloud_coverage_percent SMALLINT NOT NULL,
+    visibility_m SMALLINT NOT NULL,
+    at TIMESTAMP NOT NULL,
+    PRIMARY KEY (forecast_id),
+    FOREIGN KEY (county_id) REFERENCES county(county_id)
+);
+
+CREATE TABLE subscriber (
+    subscriber_id BIGINT GENERATED ALWAYS AS IDENTITY,
+    subscriber_username VARCHAR(30) NOT NULL,
```
Comment (MurasatoNaoya) [inline database/schema.sql:96]: Nice varchar ranges, as we can reasonably expect subscriber username, phone number and email to be within a specific range.

File: database/schema.sql [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    county_name VARCHAR(35) NOT NULL,
+    latitude FLOAT NOT NULL,
+    longitude FLOAT NOT NULL,
+    region_id SMALLINT NOT NULL,
+    PRIMARY KEY (county_id),
+    FOREIGN KEY (region_id) REFERENCES region(region_id)
+);
+
+CREATE TABLE solar_feature (
+    sunrise_id BIGINT GENERATED ALWAYS AS IDENTITY,
```
Comment (AlexOBarnes) [inline database/schema.sql:73]:
```suggestion
    feature_id BIGINT GENERATED ALWAYS AS IDENTITY,
```

File: database/schema.sql [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+);
+
+CREATE TABLE subscriber (
+    subscriber_id BIGINT GENERATED ALWAYS AS IDENTITY,
+    subscriber_username VARCHAR(30) NOT NULL,
+    subscriber_phone VARCHAR(15),
+    subscriber_email VARCHAR(255),
+    PRIMARY KEY (subscriber_id),
+    CONSTRAINT phone_or_email_exist CHECK (
+        subscriber_phone IS NOT NULL OR subscriber_email IS NOT NULL
```
Comment (MurasatoNaoya) [inline database/schema.sql:101]: Nice conditional check for ensuring that a subscriber has provided at least their phone number or email.

File: database/schema.sql [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+);
+
+CREATE TABLE body_assignment (
+    assignment_id BIGINT GENERATED ALWAYS AS IDENTITY,
+    region_id SMALLINT NOT NULL,
+    body_id BIGINT NOT NULL,
+    begin_time TIMESTAMP NOT NULL,
+    end_time TIMESTAMP NOT NULL,
+    latitude FLOAT NOT NULL,
+    longitude FLOAT NOT NULL,
```
Comment (AlexOBarnes) [inline database/schema.sql:56]: Should we specify how long the float should be?

File: database/schema.sql [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+    aurora_colour_id SMALLINT,
+    colour VARCHAR(6) NOT NULL,
+    description VARCHAR(30) NOT NULL,
+    meaning VARCHAR(255) NOT NULL,
+    PRIMARY KEY (aurora_colour_id)
+);
+
+CREATE TABLE aurora_alert (
+    alert_id BIGINT GENERATED ALWAYS AS IDENTITY,
+    alert_time TIMESTAMP NOT NULL,
```
Comment (AlexOBarnes) [inline database/schema.sql:21]:
```suggestion
    alert_time TIMESTAMP NOT NULL CHECK(alert_time<=CURRENT_TIMESTAMP),
```

**Review by MurasatoNaoya — APPROVED** [review]
A good schema.sql script for defining the ERD we have all agreed on for StarWatch.

File: database/schema.sql [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+    county_id SMALLINT NOT NULL,
+    temperature_c FLOAT NOT NULL,
+    precipitation_probability_percent SMALLINT NOT NULL,
+    precipitation_mm FLOAT NOT NULL,
+    cloud_coverage_percent SMALLINT NOT NULL,
+    visibility_m SMALLINT NOT NULL,
+    at TIMESTAMP NOT NULL,
+    PRIMARY KEY (forecast_id),
+    FOREIGN KEY (county_id) REFERENCES county(county_id)
+);
```
Comment (AlexOBarnes) [inline database/schema.sql:92]: This looks good I like the use of units

File: database/schema.sql [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+CREATE TABLE subscriber (
+    subscriber_id BIGINT GENERATED ALWAYS AS IDENTITY,
+    subscriber_username VARCHAR(30) NOT NULL,
+    subscriber_phone VARCHAR(15),
+    subscriber_email VARCHAR(255),
+    PRIMARY KEY (subscriber_id),
+    CONSTRAINT phone_or_email_exist CHECK (
+        subscriber_phone IS NOT NULL OR subscriber_email IS NOT NULL
+    )
+);
```
Comment (AlexOBarnes) [inline database/schema.sql:103]: Very elegant check

**Review by AlexOBarnes — CHANGES_REQUESTED** [review]

File: database/schema.sql (hunk 2 above)
Comment (olliejth) [inline database/schema.sql:1]: This could be done with separate queries for clarity?

**Review by olliejth — APPROVED** [review]
Descriptive table names

**Comment by AlexOBarnes** [comment]
What wonderfully consistent table names

File: database/schema.sql (hunk 9 above)
Comment (CalebMathers) [reply database/schema.sql:56]: Lat/Lon will be calculated from the azimuth so for now we cannot predict the lengths.

**Review by AlexOBarnes — APPROVED** [review]
Thanks for making these changes

**Pull request merged**